src/utilities/filetypes/AlfalfaJSON.cpp

```diff
@@ -0,0 +1,223 @@
+                                                                  const std::string& display_name) {
+    AlfalfaOutputVariable component(variable_key, variable_name);
+    return exposeFromComponent(component, display_name);
+  }
+
+  boost::optional<AlfalfaPoint> AlfalfaJSON::exposeGlobalVariable(const std::string& variable_name, const std::string& display_name) {
+    AlfalfaGlobalVariable component(variable_name);
+    return exposeFromComponent(component, display_name);
+  }
+
+  void AlfalfaJSON::exposePoint(const AlfalfaPoint& point) {
+    m_impl->exposePoint(point);
+  }
+
+  boost::optional<AlfalfaPoint> AlfalfaJSON::exposeFromObject(const openstudio::IdfObject& idf_object, const std::string& display_name) {
+    IddObjectType idd_type = idf_object.iddObject().type();
+    std::string _display_name = display_name;
+
+    if (display_name.size() == 0) {
+      boost::optional<std::string> name = getName(idf_object);
+      if (name.is_initialized()) {
+        _display_name = name.get();
+      }
+    }
+
+    AlfalfaComponent* component;
+    try {
+      switch (idd_type.value()) {
+        case IddObjectType::OS_Output_Meter:
+        case IddObjectType::Output_Meter:
+          component = new AlfalfaMeter(idf_object);
+          break;
+        case IddObjectType::OS_EnergyManagementSystem_Actuator:
+        case IddObjectType::EnergyManagementSystem_Actuator:
+          component = new AlfalfaActuator(idf_object);
+          break;
+        case IddObjectType::OS_Output_Variable:
+        case IddObjectType::Output_Variable:
+        case IddObjectType::EnergyManagementSystem_OutputVariable:
+        case IddObjectType::OS_EnergyManagementSystem_OutputVariable:
+          component = new AlfalfaOutputVariable(idf_object);
+          break;
+        case IddObjectType::OS_EnergyManagementSystem_GlobalVariable:
+        case IddObjectType::EnergyManagementSystem_GlobalVariable:
+          component = new AlfalfaGlobalVariable(idf_object);
+          break;
+        default:
+          LOG(Error, "Unable to create Alfalfa Point from Object of type " + idd_type.valueDescription());
+          return boost::none;
+      }
+    } catch (std::runtime_error& error) {
+      LOG(Error, error.what());
+      return boost::none;
+    }
+    return exposeFromComponent(*component, _display_name);
+  }
+
+  boost::optional<AlfalfaPoint> AlfalfaJSON::exposeFromComponent(const AlfalfaComponent& component, const std::string& display_name) {
+    std::string _display_name = display_name;
+    if (display_name.size() == 0) {
+      if (component.type == "Actuator") {
+        _display_name = "Actuator for " + component.parameters["component_name"].asString() + ":" + component.parameters["component_type"].asString()
+                        + ":" + component.parameters["control_type"].asString();
+      } else if (component.type == "Constant") {
+        LOG(Error, "Constant points must be provided with a display name");
+        return boost::none;
```

Comment:
There's something bothering me with error handling here, or maybe public-ness of the exposeFromComponent method.

Why ever allow a component.type Constant to NOT have a displayName to begin with?!

Reply:
It was mostly in the interest of making the base programming interface as simple as possible for users to get up and running with. Maybe I went too far.

Reply:
this has now been improved